components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -428,9 +428,11 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
                     } else {
                         Map<String, String> paramMap = new HashMap<>();
                         paramMap.put(BasicAuthenticatorConstants.ERROR_CODE, errorCode);
-                        paramMap.put(BasicAuthenticatorConstants.FAILED_USERNAME,
-                                URLEncoder.encode(request.getParameter(BasicAuthenticatorConstants.USER_NAME),
-                                        BasicAuthenticatorConstants.UTF_8));
+                        if (request.getParameter(BasicAuthenticatorConstants.USER_NAME) != null){
+                            paramMap.put(BasicAuthenticatorConstants.FAILED_USERNAME,
+                                    URLEncoder.encode(request.getParameter(BasicAuthenticatorConstants.USER_NAME),
+                                            BasicAuthenticatorConstants.UTF_8));
+                        }
```

Comment:
If the username is null, don't need an exception or at least a debug log here?

cause earlier flow will assume Failed username will be there in param map